GitUI/Shells/IShellDescriptor.cs

```diff
@@ -0,0 +1,46 @@
+﻿#nullable enable
+
+using System.Drawing;
+
+namespace GitUI.Shells
+{
+    public interface IShellDescriptor
+    {
+        /// <summary>
+        /// Gets the executable command line.
+        /// </summary>
+        public string? ExecutableCommandLine { get; }
+
+        /// <summary>
+        /// Gets the executable name.
+        /// </summary>
+        public string ExecutableName { get; }
+
+        /// <summary>
+        /// Gets the executable resolved path.
+        /// </summary>
+        public string? ExecutablePath { get; }
+
+        /// <summary>
+        /// Indicates whether the executable for the given shell is located and the shell can be run.
+        /// </summary>
+        public bool HasExecutable { get; }
+
+        /// <summary>
+        /// Gets the shell icon.
+        /// </summary>
+        public Image Icon { get; }
+
+        /// <summary>
+        /// Gets the user visible shell name.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Gets the command to execute when a new terminal starts.
```

Comment:
The comment needs rewording
```suggestion
        /// Gets the command to execute when a new terminal starts, or a working directory is switched.
```

Reply:
Yep, even remove the first part. It's a different kettle of fish that the command is executed on terminal start, too.